utils/src/foreign_field.rs

```diff
@@ -0,0 +1,121 @@
+use std::fmt::Display;
+
+use crate::field_helpers::FieldHelpers;
+use ark_ff::FftField;
+use num_bigint::BigUint;
+
+pub const LIMB_BITS: u32 = 88;
+
+pub const LIMB_COUNT: usize = 3;
+
+/// The foreign field modulus of secp256k1 is the prime number (in big endian)
+/// FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFE FFFFFC2F
+/// given by the computation 2^256 - 2^32 - 2^9 - 2^8 - 2^7 - 2^6 - 2^4 - 1
+/// more information here  https://en.bitcoin.it/wiki/Secp256k1
+pub const FOREIGN_MOD: &[u8] = &[
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFE, 0xFF, 0xFF, 0xFC, 0x2F,
+];
+pub const FOREIGN_BITS: usize = 8 * FOREIGN_MOD.len(); // 256 bits
+
+#[derive(Debug, Clone, Copy)]
+pub struct ForeignElement<F, const N: usize> {
+    /// limbs in little endian order
+    pub limbs: [F; N],
+    /// number of limbs used for the foreign field element
+    pub len: usize,
+}
+
+impl<F: FftField, const N: usize> ForeignElement<F, N> {
+    /// Initializes a new foreign element from a big unsigned integer
+    pub fn new_from_big(big: BigUint) -> Self {
+        let vec = ForeignElement::<F, N>::big_to_vec(big);
+
+        // create an array of N native elements containing the limbs
+        // until the array is full in big endian, so most significant
+        // limbs may be zero if the big number is smaller
+        let limbs = {
+            let mut ini = [F::zero(); N];
+            for i in 0..N {
+                if i < vec.len() {
```

Comment:
I pointed that in the other PR, but if vec.len() > N?

Reply:
The current version panics, but we can be more elegant here.

Reply:
I'm not sure I understand where it would panic.

(Also in general it's not good practice to do `for i in 0..N`, it's better to iterate through an iterator)

Reply:
Mmm right now it panics in the FFAdd branch if we were to instantiate a foreign_field_modulus parameter that does not fit in 3 limbs -> [here](https://github.com/o1-labs/proof-systems/blob/21feaf439fc0c7b353fd35cae93d5c3288e3f9f9/kimchi/src/circuits/constraints.rs#L394)